patches/api/0483-Vault-API.patch

```diff
@@ -0,0 +1,97 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Boy <sivertpaulsen2@gmail.com>
+Date: Wed, 24 Jul 2024 13:41:32 +0200
+Subject: [PATCH] Vault API
+
+
+diff --git a/src/main/java/org/bukkit/block/Vault.java b/src/main/java/org/bukkit/block/Vault.java
+index 375f41fe4aebdbf140497b5ba1e54e1c72f0cdfc..035a2104d878bbde0226b3234b334080159d8038 100644
+--- a/src/main/java/org/bukkit/block/Vault.java
++++ b/src/main/java/org/bukkit/block/Vault.java
+@@ -1,9 +1,63 @@
+ package org.bukkit.block;
+ 
+-import org.jetbrains.annotations.ApiStatus;
++import net.kyori.adventure.key.Key;
++import org.bukkit.entity.Player;
++import org.bukkit.inventory.ItemStack;
++import org.bukkit.loot.LootTable;
++import org.jetbrains.annotations.Nullable;
++import java.util.Collection;
++import java.util.List;
++import java.util.UUID;
+ 
+ /**
+- * Represents a captured state of a trial spawner.
++ * Represents a captured state of a vault.
+  */
+ public interface Vault extends TileState {
++
++    // Paper start - Vault API
++
++    // Vault Config
++    public LootTable getLootTable();
++
++    public void setLootTable(LootTable lootTable);
++
++    @Nullable
++    public LootTable getOverrideDisplayLootTable();
++
++    public void setOverrideDisplayLootTable(@Nullable LootTable lootTable);
++
++    public ItemStack getKeyItem();
++
++    public void setKeyItem(ItemStack item);
++
++    public double getActivationRange();
++
++    public void setActivationRange(double activationRange);
++
++    public double getDeactivationRange();
++
++    public void setDeactivationRange(double deactivationRange);
++
++    // Vault ServerData
++    public Collection<Player> getRewardedPlayers();
++
++    public Collection<UUID> getRewardedPlayerUUIDs();
++
++    public boolean hasRewardedPlayer(Player player);
++
++    public boolean hasRewardedPlayer(UUID uniqueId);
++
++    public void addToRewardedPlayers(Player player);
++
++    public void addToRewardedPlayers(UUID uniqueId);
++
++    public void removeFromRewardedPlayers(Player player);
++
++    public void removeFromRewardedPlayers(UUID uniqueId);
++
++    public List<ItemStack> getItemsToEject();
++
++    public void setItemsToEject(List<ItemStack> itemsToEject);
++
++    // Paper end - Vault API
+ }
+diff --git a/src/main/java/org/bukkit/block/data/type/Vault.java b/src/main/java/org/bukkit/block/data/type/Vault.java
+index 87e46dfb951b6e4b753af7c31efd444e549052b9..9a8ee2bcd1b3bd89d11aab87bb5153f0d8e88458 100644
+--- a/src/main/java/org/bukkit/block/data/type/Vault.java
++++ b/src/main/java/org/bukkit/block/data/type/Vault.java
+@@ -16,14 +16,14 @@ public interface Vault extends Directional {
+      * @return the 'vault_state' value
+      */
+     @NotNull
+-    State getTrialSpawnerState();
```

Comment:
This might break plugins. Maybe keep those old methods but deprecate them

Reply:
In my opinion this is a fine api break, considering this is clearly a wrong name and the methods have not been around for long and since the naming of these methods might cause for confusion.

Reply:
While I don't hate the API break here, we cannot break ABI for spigot plugins as those will need to use said methods.
As such, this PR would have to address the conversion in commodore byte code rewriting.

Reply:
That part can be remove when pr is rebased. see https://hub.spigotmc.org/stash/projects/SPIGOT/repos/bukkit/commits/8a9ecf29405887b9e58244cd5d87e8ca0a5970d4#src%2Fmain%2Fjava%2Forg%2Fbukkit%2Fblock%2Fdata%2Ftype%2FVault.java